jb/itch5/order_book.hpp

```diff
@@ -174,11 +220,8 @@ class order_book {
   }
 
 private:
-  typedef std::map<price4_t, int, std::greater<price4_t>> buys;
-  typedef std::map<price4_t, int, std::less<price4_t>> sells;
-
-  buys buy_;
-  sells sell_;
+  typename book_type::buys_t buy_;
```

Comment:
Some alternatives to consider:

template<typename buys_t, typename sells_t>
class order_book {
};

or even:

template<template<typename compare> class book_side_type>
class order_book {
/* stuff */
private:
  book_side_type<std::greater> ...;
  book_side_type<std::less> ...;
};